mlflow/models/utils.py

```diff
@@ -208,15 +240,24 @@ def _handle_dataframe_input(df):
                     f"but got '{type(input_example)}'",
                 )
             self.data = _handle_dataframe_input(self._inference_data)
-            self.info = {
-                "artifact_path": example_filename,
-                "type": "dataframe",
-                "pandas_orient": "split",
-            }
+            self.info.update(
+                {
+                    "type": "dataframe",
+                    "pandas_orient": "split",
+                }
+            )
 
     def save(self, parent_dir_path: str):
         """Save the example as json at ``parent_dir_path``/`self.info['artifact_path']`."""
-        with open(os.path.join(parent_dir_path, self.info["artifact_path"]), "w") as f:
+        if self._inference_params is not None:
+            if "params" in self.data:
```

Comment:
Should we externalize this constant since it is referred within this utils module as well as in `signature.py`? 

I'm thinking that a more unique reserved key (similar to our reserved keys within tags) name might be less likely to cause an issue with users. `"mlflow.model.params"` or something like that is highly unlikely to be used by an end user and could be seen purely as the internal key reference for handling this data. 

WDYT?

Reply:
Using a reserved name is NOT a blocker for this PR. 

At a minimum though, even if we stick with the "params" reserved naming convention (which is fine), I do feel like this should be a constant defined within the utils module so that each referring location can have a standard definition for this key (just to simplify future modifications) :)

Reply:
Yea that makes sense :)